packages/design-system/src/components/ButtonIcon/ButtonIcon.stories.mdx

```diff
@@ -0,0 +1,177 @@
+
+#### Toggle
+
+If the button needs to hold an active / inactive state, then it is a Toggle.
+
+Don’t use Toggles in series, opt for badge dropdowns instead.
+
+<Canvas>
+	<Story story={Stories.Toggle} name="ButtonToggle" />
+</Canvas>
+
+<ArgsTable story="ButtonToggle" />
+
+#### Floating
+
+This variant is only used when the button floats on top of content.
+
+It could be it's sitting on top of a line connecting two nodes, or sticky on top of a draggable scene for instance.
+
+<Canvas>
+	<Story story={Stories.Floating} name="ButtonIconFloating" />
+</Canvas>
+
+<ArgsTable story="ButtonIconFloating" />
+
+## States
+
+ButtonIcons can display a "loading" state for asynchronous actions.
+
+### Loading
+
+Only use when necessary to avoid multiple clicks when the action is not instant.
+
+<Canvas>
+	<Story story={Stories.Loading} />
+</Canvas>
+
+### Skeleton
+
+Skeleton needs are handled by the Skeleton component.
+
+// TODO
+
+## Interaction
+
+All buttons have interactive states for "hover", "active" and "disabled". A focus ring should also be displayed on keyboard navigation.
+
+## Content
+
+All ButtonIcons carry a tooltip on hover. The content of that tooltip should be short ("do something") and effective.
+
+## Usage
+
+You have access to three aliases for variants with curated props for each, but you may want to use a single component and switch between variants.
+
+You can:
+
+```tsx
+import { ButtonIconVariant } from '@talend/design-system';
+
+function MyComponent() {
+	return (
+		<>
+			<ButtonIconVariant variant="default" size="XS" onClick={onAddDataset()}>
+				Add a dataset
+			</ButtonIconVariant>
+			<ButtonIconVariant variant="toggle" isActive size="S" onClick={toggleAdvancedMode()}>
+				Turn Advanced Mode off
+			</ButtonIconVariant>
+			<ButtonIconVariant variant="floating" onClick={onZoomIn()} disabled>
+				Zoom in
+			</ButtonIconVariant>
+		</>
+	);
+}
+```
+
+<Canvas>
+	<Story
+		name="Standalone with variant prop"
+		args={{
+			variant: 'default',
+			icon: 'talend-datagrid',
+			children: 'Action label',
+			onClick: () => {},
+		}}
+		argTypes={{
+			variant: {
+				description: 'ButtonIcon variation',
+				control: {
+					type: 'select',
+					options: ['default', 'toggle', 'floating'],
+				},
+			},
+			icon: {
+				description: 'Try it out with another talend icon you know',
+				control: {
+					type: 'text',
+				},
+			},
+			children: {
+				description: 'Contents of the tooltip',
+				control: {
+					type: 'text',
+				},
+			},
+		}}
+	>
+		{({ variant, ...rest }) => {
+			const Component = ButtonIconVariant;
+			Component.displayName = 'ButtonIconVariant';
+			return <Component variant={variant} {...rest} />;
+		}}
+	</Story>
+</Canvas>
+
+<ArgsTable story="Standalone with variant prop" />
+
+Of course all buttons can also use natural `<button>` attributes (but not `classNames` nor `style`).
+
+<Canvas>
+	<Story story={Stories.NaturalButtonProps} />
+</Canvas>
+
+## Accessibility
+
+N/C
```

Comment:
A word about aria-pressed attr maybe?

Reply:
Sure - it's transparent to the user but why not